lib/bpf/bpf_test.go

```diff
@@ -150,66 +150,64 @@ func TestRootObfuscate(t *testing.T) {
 	}
 
 	// Find the programs needed to run these tests on the host.
-	decoderPath, err := os_exec.LookPath("base64")
+	decoderPath, err := osexec.LookPath("base64")
 	require.NoError(t, err)
-	shellPath, err := os_exec.LookPath("sh")
+	shellPath, err := osexec.LookPath("sh")
 	require.NoError(t, err)
 
 	// Start execsnoop.
 	execsnoop, err := startExec(8192)
 	defer execsnoop.close()
 	require.NoError(t, err)
 
-	// Create a context that will be used to signal that an event has been received.
-	doneContext, doneFunc := context.WithCancel(context.Background())
+	// Create obfuscated script.
+	shellContents := fmt.Sprintf("#!%v\necho bHM= | %v --decode | %v",
+		shellPath, decoderPath, shellPath)
 
-	// Start two goroutines. The first writes a script which will execute "ls"
-	// in a loop. The second waits for an exec event to show up the reports "ls"
+	// Write script to a temporary folder.
+	fileName := filepath.Join(t.TempDir(), "test-script")
+	err = os.WriteFile(fileName, []byte(shellContents), 0700)
+	require.NoError(t, err)
+
+	done := make(chan struct{})
```

Comment:
It feels like there's a good amount of repetition here that could be functionalized. Maybe extract some of these loops/`execsnoop` calls into their own functions?

Reply:
I'm only looking to fix the correctness of tests here, not rewrite them.

In general, if you don't feel strongly enough about your suggestions to block merge, just leave a command and don't request changes. Otherwise we have to chase you down to get you to remove the request changes state.